draft

en opened the pull request.

Satoshi-Kusumoto commented on spec/ics-010-grandpa-client/README.md:
```diff
@@ -0,0 +1,371 @@
+
+## Synopsis
+
+(high-level description of and rationale for specification)
```
```suggestion
```

Satoshi-Kusumoto commented on spec/ics-010-grandpa-client/README.md:
```diff
@@ -0,0 +1,371 @@
+
+### Motivation
+
+(rationale for existence of standard)
```
```suggestion
```

Satoshi-Kusumoto commented on spec/ics-010-grandpa-client/README.md:
```diff
@@ -0,0 +1,371 @@
+
+### Definitions
+
+(definitions of any new terms not defined in common documentation)
```
```suggestion
```

Satoshi-Kusumoto commented on spec/ics-010-grandpa-client/README.md:
```diff
@@ -0,0 +1,371 @@
+
+### Desired Properties
+
+(desired characteristics / properties of protocol, effects if properties are violated)
```
```suggestion
```

Satoshi-Kusumoto commented on spec/ics-010-grandpa-client/README.md:
```diff
@@ -0,0 +1,371 @@
+
+## Technical Specification
+
+(main part of standard document - not all subsections are required)
```
```suggestion
```

Satoshi-Kusumoto commented on spec/ics-010-grandpa-client/README.md:
```diff
@@ -0,0 +1,371 @@
+
+(main part of standard document - not all subsections are required)
+
+(detailed technical specification: syntax, semantics, sub-protocols, algorithms, data structures, etc)
```
```suggestion
```

Satoshi-Kusumoto commented on spec/ics-010-grandpa-client/README.md:
```diff
@@ -0,0 +1,371 @@
+### Data Structures
+
+(new data structures, if applicable)
+
```
```suggestion
```

Satoshi-Kusumoto commented on spec/ics-010-grandpa-client/README.md:
```diff
@@ -0,0 +1,371 @@
+(main part of standard document - not all subsections are required)
+
+(detailed technical specification: syntax, semantics, sub-protocols, algorithms, data structures, etc)
+This specification depends on correct instantiation of the [GRANDPA finality gadget](https://github.com/w3f/consensus/blob/master/pdf/grandpa.pdf).
```
```suggestion
This specification depends on correct instantiation of the [GRANDPA finality gadget](https://github.com/w3f/consensus/blob/master/pdf/grandpa.pdf) and its light client algorithm.
```

Satoshi-Kusumoto commented on spec/ics-010-grandpa-client/README.md:
```diff
@@ -0,0 +1,371 @@
+### Sub-protocols
+
+(sub-protocols, if applicable)
+
```
```suggestion
```

Satoshi-Kusumoto commented on spec/ics-010-grandpa-client/README.md:
```diff
@@ -0,0 +1,371 @@
+
+### Properties & Invariants
+
+(properties & invariants maintained by the protocols specified, if applicable)
```
```suggestion
```

Satoshi-Kusumoto commented on spec/ics-010-grandpa-client/README.md:
```diff
@@ -0,0 +1,371 @@
+stage: draft
+category: IBC/TAO
+kind: instantiation
+author: Yuanchao Sun <ys@cdot.network>
```
```suggestion
author: Yuanchao Sun <ys@cdot.network>, John Wu <john@cdot.network>
```

Satoshi-Kusumoto reviewed (changes requested):
Please add the explanation of setId, and organize a discussion about the justification scheme

en committed:
ICS 10: GRANDPA client

This is a initial spec for GRANDPA finality gadget.
In future versions, we will add a verification for TrustPeriod if
necessary and provide pseudocode for verifying justification.
